enhancements/ingress/set-replace-append-delete-deny-headers.md

```diff
@@ -0,0 +1,906 @@
+via route annotation.
+
+### User Stories
+
+1. As a cluster administrator, I need to configure an IngressController to set or delete certain HTTP request or response headers in order to
+   satisfy compliance requirements at the global level (i.e. for all routes) to send or not to send those particular headers to clients.
+   Example 1: As a cluster administrator, I want to set HTTP response headers `X-XSS-Protection: 1; mode=block`, `X-Content-Type-Options: nosniff`, and `X-Frame-Options: SAMEORIGIN` to be sent to all clients.
+   Example 2: As a cluster administrator I want to set HTTP request headers such as `Content-Type: text/html; charset=utf-8`, `X-SSL-Client-Cert: %{+Q}[ssl_c_der,base64]`, or `X-Target: %[req.hdr(host),lower]` (using HAProxy's  dynamic value syntax) to be sent to all applications.
+   Example 3: As a cluster administrator I want to delete HTTP response headers such as `server` to prevent them from being sent to the client.
+   Example 4: As a cluster administrator I want to delete http request headers like `Content-Type` to be not sent to the application.
+2. As a route owner, I need to configure my Route to set or delete HTTP request or response headers to send or not send them to the client which has connected to the Route.
+   Example 1: As a user, I want to set HTTP response headers such as `X-XSS-Protection: 1; mode=block`, `X-Content-Type-Options: nosniff`, and `X-Frame-Options: SAMEORIGIN` to be sent to the client.
+   Example 2: As a user, I want to set HTTP request headers such as `X-SSL-Client-Cert: %{+Q}[ssl_c_der,base64]` and `X-Target: %[req.hdr(host),lower]` (using HAProxy's  dynamic value syntax) to be sent to the application.
+   Example 3: As a user, I want to delete HTTP response headers such as `X-XSS-Protection` to prevent them from being sent to the client.
+   Example 4: As a user I want to delete HTTP request headers such as `server` to prevent them from being sent to the application.
+
+### Goals
+
+1. Enable the cluster administrator to specify HTTP response headers in the IngressController CR spec that should be set or
+   deleted at the global level when forwarding a response to all clients, i.e. for all routes.
+2. Enable the cluster administrator to specify HTTP request headers in the IngressController CR spec that should be set or
+   deleted at the global level when forwarding connections to all applications from clients.
+3. Enable the route owner to specify HTTP response headers in the Route spec (when it is not a passthrough Route) for their application
+   that should be set or deleted when forwarding a response to the client.
+4. Enable the route owner to specify HTTP request headers in the Route spec (when it is not a passthrough Route) for their application
+   that should be set or deleted when forwarding a connection to the application from a client.
+5. Define an immutable/hard-coded deny list of headers which are not allowed to be set or deleted.
```

Comment:
Is HSTS is the only entry in the deny list?  I feel like we would also need to include any of the headers that we specifically hard-code in the haproxy config template as I mentioned in:
https://github.com/openshift/router/pull/438/files#r1167340046
https://github.com/openshift/router/pull/438/files#r1167340360

But that is not the whole list.